addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/QueuedSend.java

```diff
@@ -0,0 +1,344 @@
+
+            // The the next pointer of the last element in the chain to the new item
+            lastInChain.next = newItem;
+
+            // Set the pointer to the last element in the chain to the new command
+            root.last = newItem;
+
+            // Always return the root
+            return this.root;
+        }
+
+        public QueueItem addNextRepeatable(byte[] data) {
+            addNext(data);
+            root.last.repeatable = true;
+            return this.root;
+        }
+    }
+
+    BlockingQueue<QueueItem> queue = new LinkedBlockingQueue<>(20);
+    protected final DatagramPacket packet;
+    protected final DatagramSocket datagramSocket;
+    private static final Logger logger = LoggerFactory.getLogger(QueuedSend.class);
+    private int delay_between_commands = 100;
+    private int repeat_commands = 1;
+    private boolean willbeclosed = false;
+    private Thread thread;
+
+    public static final byte NO_CATEGORY = 0;
+
+    /**
+     * Creates a new send queue and starts the background thread. Call setAddress and
+     * setPort before using any of the queue commands.
+     *
+     * @throws SocketException
+     */
+    public QueuedSend() throws SocketException {
+        byte[] a = new byte[0];
+        packet = new DatagramPacket(a, a.length);
+        datagramSocket = new DatagramSocket();
+    }
+
+    /**
+     * Start the send thread of this queue. Call dispose() to quit the thread.
+     */
+    public void start() {
+        willbeclosed = false;
+        thread = new Thread(this);
+        thread.start();
+    }
+
+    public int getDelayBetweenCommands() {
+        return delay_between_commands;
+    }
+
+    public int getRepeatCommands() {
+        return repeat_commands;
+    }
+
+    public void setRepeatCommands(int repeat_commands) {
+        repeat_commands = Math.max(0, Math.min(5, repeat_commands));
+        this.repeat_commands = repeat_commands;
+    }
+
+    public void setDelayBetweenCommands(int ms) {
+        ms = Math.max(0, Math.min(400, ms));
+        delay_between_commands = ms;
+    }
+
+    /**
+     * The queue process
+     */
+    @Override
+    public void run() {
+        QueueItem item = null;
+        while (!willbeclosed) {
+            int delay_time = delay_between_commands;
+
+            // If the command belongs to a chain of commands, get the next command now.
+            if (item != null && item.next != null) {
+                item = item.next;
+            } else {
+                try {
+                    // block/wait for another item
+                    item = queue.take();
+                } catch (InterruptedException e) {
+                    if (!willbeclosed) {
+                        logger.error("Queue take failed: " + e.getLocalizedMessage());
+                    }
+                    break;
+                }
+            }
+
+            if (item == null || item.unique_command_id == QueueItem.INVALID) {
+                continue;
```

Comment:
Only the first item in a chain is invalidated so when you call continue it will send the second command.

Should you be setting item to null if the start of the chain is invalid?